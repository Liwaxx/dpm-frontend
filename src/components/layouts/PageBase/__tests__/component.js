import React from 'react';
import renderer from 'react-test-renderer';
import Component from '../component';

jest.mock('@material-ui/core/Snackbar');
jest.mock('../../../elements/Text');
jest.mock('../../../elements/DrawerList');
jest.mock('../../../../constants/drawerListData');
jest.mock('.././../../../utils/common');

describe('Component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Component actions={{}} classes={{}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with snackbar', () => {
    const tree = renderer
      .create(<Component actions={{}} classes={{}} message="test" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});