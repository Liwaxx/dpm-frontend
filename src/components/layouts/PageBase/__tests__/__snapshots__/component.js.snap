// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<div>
  <div>
    <div
      className="MuiDrawer-docked-1"
    >
      <div
        className="MuiPaper-root-12 MuiPaper-elevation0-14 MuiDrawer-paper-2 MuiDrawer-paperAnchorLeft-3 MuiDrawer-paperAnchorDockedLeft-7"
      >
        <div />
        <div />
      </div>
    </div>
  </div>
  <div>
    <div
      className="MuiDrawer-docked-1"
    >
      <div
        className="MuiPaper-root-12 MuiPaper-elevation0-14 MuiDrawer-paper-2 MuiDrawer-paperAnchorLeft-3 MuiDrawer-paperAnchorDockedLeft-7"
      >
        <ul
          className="MuiList-root-39 MuiList-padding-40"
        />
      </div>
    </div>
  </div>
  <main />
</div>
`;

exports[`Component renders with snackbar 1`] = `
<div>
  <div>
    <div
      className="MuiDrawer-docked-1"
    >
      <div
        className="MuiPaper-root-12 MuiPaper-elevation0-14 MuiDrawer-paper-2 MuiDrawer-paperAnchorLeft-3 MuiDrawer-paperAnchorDockedLeft-7"
      >
        <div />
        <div />
      </div>
    </div>
  </div>
  <div>
    <div
      className="MuiDrawer-docked-1"
    >
      <div
        className="MuiPaper-root-12 MuiPaper-elevation0-14 MuiDrawer-paper-2 MuiDrawer-paperAnchorLeft-3 MuiDrawer-paperAnchorDockedLeft-7"
      >
        <ul
          className="MuiList-root-39 MuiList-padding-40"
        />
      </div>
    </div>
  </div>
  <main />
</div>
`;
